cmd/snap/cmd_model.go

```diff
@@ -94,25 +98,40 @@ func init() {
 }
 
 func (x *cmdModel) Execute(args []string) error {
-	var assertion asserts.Assertion
-	var err error
-	if x.Serial {
-		assertion, err = x.client.CurrentSerialAssertion()
-	} else {
-		assertion, err = x.client.CurrentModelAssertion()
-	}
+	var mainAssertion asserts.Assertion
+	serialAssertion, serialErr := x.client.CurrentSerialAssertion()
+	modelAssertion, modelErr := x.client.CurrentModelAssertion()
 
-	if err != nil {
-		if client.IsAssertionNotFoundError(err) {
+	// if we didn't get a model assertion bail early
+	if modelErr != nil {
+		if client.IsAssertionNotFoundError(modelErr) {
 			// device is not registered yet - use specific error message
-			err = fmt.Errorf(i18n.G("device not ready - no assertion found"))
+			modelErr = fmt.Errorf(devNotReadyMessage)
 		}
-		return err
+		return modelErr
+	}
+
+	// if the serial assertion error is anything other than not found, also
+	// bail early
+	// the serial assertion not being found may not be fatal
+	if serialErr != nil && !client.IsAssertionNotFoundError(serialErr) {
+		return serialErr
+	}
+
+	if x.Serial {
+		mainAssertion = serialAssertion
+	} else {
+		mainAssertion = modelAssertion
 	}
 
-	// --assertion means output the raw assertion
 	if x.Assertion {
-		_, err = Stdout.Write(asserts.Encode(assertion))
+		// if we are using the serial assertion and we specifically didn't find the
+		// serial assertion, bail with specific error
+		if x.Serial && client.IsAssertionNotFoundError(serialErr) {
+			return errors.New(devNotReadyMessage)
```

Comment:
I think, this should say  "device not registered yet (no serial assertion found)"

Reply:
okay changed